### PR TITLE
Fixes for esp idf commit 79f206be47c3f608615c1de8c491107e6c9194bb and…

### PR DESCRIPTION
**Pull request by Philius**
… start of esp-idf integration.

I was in the process of starting esp-idf integration when I checked for updates to esp-idf.
The changes were simple so I included them in this pull request.

### CODE REVIEW

File: cpp_utils/BLEClient.h [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #include <string.h>
 #include <map>
 #include <string>
-#include <BLERemoteService.h>
```
Comment (nkolban) [inline cpp_utils/BLEClient.h]: Good change

File: cpp_utils/BLEUtils.cpp [hunk 2]
```diff
@@ -1286,8 +1286,7 @@ void BLEUtils::dumpGattClientEvent(
 		// - uint16_t          conn_id
 		// - esp_bd_addr_t     remote_bda
 		case ESP_GATTC_CONNECT_EVT: {
-			ESP_LOGD(LOG_TAG, "[staus: %s, conn_id: %d, remote_bda: %s]",
-				BLEUtils::gattStatusToString(evtParam->connect.status).c_str(),
```
Comment (nkolban) [inline cpp_utils/BLEUtils.cpp]: Why remove the logging of status?

File: cpp_utils/BLEUtils.cpp [hunk 3]
```diff
@@ -1302,8 +1301,7 @@ void BLEUtils::dumpGattClientEvent(
 		// - uint16_t          conn_id
 		// - esp_bd_addr_t     remote_bda
 		case ESP_GATTC_DISCONNECT_EVT: {
-			ESP_LOGD(LOG_TAG, "[staus: %s, conn_id: %d, remote_bda: %s]",
-				BLEUtils::gattStatusToString(evtParam->disconnect.status).c_str(),
```
Comment (nkolban) [inline cpp_utils/BLEUtils.cpp]: Why remove the logging of status?

File: cpp_utils/BLEUtils.cpp [hunk 4]
```diff
@@ -1586,10 +1584,9 @@ void BLEUtils::dumpGattServerEvent(
 		} // ESP_GATTS_CONGEST_EVT
 
 		case ESP_GATTS_CONNECT_EVT: {
-			ESP_LOGD(LOG_TAG, "[conn_id: %d, remote_bda: %s, is_connected: %d]",
+			ESP_LOGD(LOG_TAG, "[conn_id: %d, remote_bda: %s]",
 				evtParam->connect.conn_id,
-				BLEAddress(evtParam->connect.remote_bda).toString().c_str(),
-				evtParam->connect.is_connected);
```
Comment (nkolban) [inline cpp_utils/BLEUtils.cpp]: Why remove the logging of is_connected?

File: cpp_utils/BLEUtils.cpp [hunk 5]
```diff
@@ -1603,10 +1600,9 @@ void BLEUtils::dumpGattServerEvent(
 		} // ESP_GATTS_CREATE_EVT
 
 		case ESP_GATTS_DISCONNECT_EVT: {
-			ESP_LOGD(LOG_TAG, "[conn_id: %d, remote_bda: %s, is_connected: %d]",
+			ESP_LOGD(LOG_TAG, "[conn_id: %d, remote_bda: %s]",
 				evtParam->connect.conn_id,
-				BLEAddress(evtParam->connect.remote_bda).toString().c_str(),
-				evtParam->connect.is_connected);
```
Comment (nkolban) [inline cpp_utils/BLEUtils.cpp]: Why remove the logging of is_connected?

File: cpp_utils/Doxyfile [hunk 6]
```diff
@@ -58,7 +58,7 @@ PROJECT_LOGO           =
 # entered, it will be relative to the location where doxygen was started. If
 # left blank the current directory will be used.
 
-OUTPUT_DIRECTORY       = /home/kolban/esp32/esptest/apps/workspace/esp32-snippets/cpp_utils/docs
```
Comment (nkolban) [inline cpp_utils/Doxyfile]: Good change

File: Kconfig [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        Select this option to use nkolban's c++ bluetooth ble wrappers.
+
+menuconfig NKOLBAN_BLE2902
+    bool "BLE2902"
+    depends on NKOLBAN_BLE
+    default y
+    help
+        Select this option to use nkolban's c++ bluetooth BLE2902 wrappers.
+
+endmenu
```
Comment (nkolban) [inline Kconfig]: Can you clarify what the intent is of these options and the impact of not having them?

File: cpp_utils/BLEUtils.cpp (hunk 2 above)
Comment (Philius) [reply cpp_utils/BLEUtils.cpp]: esp-idf removed that field in commit 79f206be47c3f608615c1de8c491107e6c9194bb.

File: cpp_utils/BLEUtils.cpp (hunk 3 above)
Comment (Philius) [reply cpp_utils/BLEUtils.cpp]: esp-idf removed that field in commit 79f206be47c3f608615c1de8c491107e6c9194bb.

File: cpp_utils/BLEUtils.cpp (hunk 4 above)
Comment (Philius) [reply cpp_utils/BLEUtils.cpp]: esp-idf removed that field in commit 79f206be47c3f608615c1de8c491107e6c9194bb.

File: cpp_utils/BLEUtils.cpp (hunk 5 above)
Comment (Philius) [reply cpp_utils/BLEUtils.cpp]: esp-idf removed that field in commit 79f206be47c3f608615c1de8c491107e6c9194bb.

File: Kconfig (hunk 7 above)
Comment (Philius) [reply Kconfig]: See component.mk

It's less than half done, but it's a start of being able to use your code in my project - I'm only using the ble part for now.

My use-case is to git-clone your repo to some local directory and create a symlink to it inside the components directory of each of my esp-idf projects that use it (1 for now).

Is there a better way to do this, outside of merging it into esp-idf?

I was going to email you about this but then that commit came along and caused compilation errors.
I thought I'd save you the time, plus they're backwards compatible :)

**Comment by nkolban** [comment]
Thank you sir!!

**Comment by Philius** [comment]
Forward, always.
Open source, for the win!

**Pull request merged**